control-plane/agents/src/bin/core/controller/scheduling/volume.rs

```diff
@@ -159,6 +187,63 @@ impl AddVolumeReplica {
     }
 }
 
+fn get_restricted_nodes(
```

Comment:
I think these 2 fn could probably move to a volume_group module

Reply:
Indeed.